enhancements/microshift/default-csi-plugin.md

```diff
@@ -0,0 +1,343 @@
+   1. If the file exists, MicroShift will read the file into memory and generate a ConfigMap containing the data under
+   the `lvmd.yaml` key.
+   1. If the file doesn't exist, MicroShift will assume hardcoded default values.
+
+    _Example ConfigMap:_
+    ```shell
+    $ oc get cm -n openshift-storage lvmd -o yaml
+    ```
+    ```yaml
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: lvmd
+      namespace: openshift-storage
+    data:
+      lvmd.yaml: |
+        device-classes:
+        - name: user-provided-config
+          volume-group: rhel
+          default: true
+        socket-name: /run/lvmd/lvmd.sock
+    ```
+
+1. MicroShift will apply the ConfigMap to the cluster.
+1. MicroShift will compute the sha256sum for the lvmd.yaml.
+1. To leverage innate Kubernetes behavior to restart lvmd, MicroShift will update an arbitrary environment variable on 
+the `topolvm-node` DaemonSet. This will result in the running lvmd pod being terminated and a new one being generated 
+in its place. This is necessary because the lvmd runtime does not watch the config file after it starts.
+
+### Risks and Mitigations
+
+### Drawbacks
+
+## Design Details
+
+### Open Questions [optional]
+
+### Test Plan
+
+[//]: # (**Note:** *Section not required until targeted at a release.*)
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+### Upgrade / Downgrade Strategy
+
+MicroShift is deployed as an application on top of an operating system, and so its upgrade/downgrade model differs
+significantly from OCP's.  Because MicroShift's cluster architecture is embedded within the MicroShift binary, 
+the entire application is upgraded as a whole, distributed by Red Hat as a set of RPMs.  The ART team will generate new
+MicroShift RPMs and make these available to users. Users deploying MicroShift on top of RHEL for Edge are provided a
+recipe for generating and deploying a new rpm-ostree.
+
+Downgrades are handled similarly, where users install an older version of MicroShift, or deploy a rpm-ostree layer
+containing the older version.
+
+Customers deploying onto RHEL for Edge will be able to rely on [Greenboot](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/composing_installing_and_managing_rhel_for_edge_images/index?extIdCarryOver=true&sc_cid=701f2000001OH7EAAW#how-are-rhel-for-edge-images-restored_managing-rhel-for-edge-images) to ensure that if an upgrade/rollback produces
+and unstable operating environment, the system will automatically roll the change back to the last known stable state.
+
+#### Upcoming API Group Changes
+
+Upstream TopoLVM is expected to change the API group of the LogicalVolume CRD to distance the API from the owning company
+(see issue topolvm/topovlm#168). A 
+[Webhook Conversion](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#webhook-conversion)
```

Comment:
Are we sure that a web hook is going to work in this case? We will have to change from one group to another, not just between versions in the same group and kind.

Reply:
The upstream changes are still in flight (see PR topolvm/topolvm#539). The upstream community has elected to not support migration.  Instead, an env var (`USE_LEGACY_PLUGIN_NAME=bool`) will be added whose purpose is to tell topovlm controllers to use the legacy or new API version for it's given cluster.  For instance, old clusters that upgrade topolvm will need to ensure their manifest for sets this `USE_LEGACY_PLUGIN_NAME=true`.  I'm not sure how much validation the plugin will do to ensure the env var value doesn't conflict with the actual state of the cluster.  It's recommended that users seeking to migrate do so manually, per volume.

Reply:
I'll go ahead and revise this statement to focus on the current state of the work.

Reply:
We will need to think about how to expose that setting for users, but we can do that in a future enhancement.